src/lib/middleware/origin-middleware.ts

```diff
@@ -0,0 +1,27 @@
+import type { Request, Response, NextFunction } from 'express';
+import type { IUnleashConfig } from '../types';
+
+export const originMiddleware = ({
+    getLogger,
+    flagResolver,
+}: Pick<IUnleashConfig, 'getLogger' | 'flagResolver'>) => {
+    const logger = getLogger('/middleware/origin-middleware.ts');
+    logger.debug('Enabling origin middleware');
+
+    return (req: Request, _: Response, next: NextFunction) => {
+        if (!flagResolver.isEnabled('originMiddleware')) {
+            return next();
+        }
+
+        const isUI = !req.headers.authorization;
+
+        if (isUI) {
+            logger.debug('UI request');
```

Comment:
That's gonna be a lotta logs! Do we need "UI request" and "API request"?

Reply:
Maybe we only log on POST/PUT/DELETE?